autoload/ale/c.vim

```diff
@@ -89,3 +89,45 @@ function! ale#c#FindCompileCommands(buffer) abort
 
     return ''
 endfunction
+
+" Given a buffer number, find a compilation database (compile_commands.json)
+" and parse this into an argument string
+function! ale#c#FindCompileArgs(buffer) abort
+    let l:compile_args = {}
+
+    let l:compile_commands_path = ale#path#FindNearestFile(a:buffer, 'compile_commands.json')
+    if !empty(l:compile_commands_path)
+        let l:bufname = fnamemodify(bufname(a:buffer), ':p')
+
+        for elem in json_decode(readfile(l:compile_commands_path))
+            if elem["file"] == l:bufname
+                let l:compile_args["directory"] = elem["directory"]
+                " skip unwanted stuff
+                "   first arg (compiler)
+                "   last arg (filename)
+                "    -o objectfile
+                "    -c
+                let l_skip = 0
+                let l:args = []
+                for arg in split(elem["command"], ' ')[2:-2]
```

Comment:
Yes. I need to create a test case locally to see how bear/cmake handle spaces in arguments and then work out how to parse whatever they output.